sdk/core/azure-core/inc/http/http.hpp

```diff
@@ -233,14 +233,17 @@ namespace Azure { namespace Core { namespace Http {
     std::string GetQueryString() const;
 
   public:
-    Request(HttpMethod httpMethod, std::string const& url, std::unique_ptr<BodyStream> bodyStream)
-        : m_method(std::move(httpMethod)), m_url(url), m_bodyStream(std::move(bodyStream)),
+    Request(HttpMethod httpMethod, std::string const& url, BodyStream& bodyStream)
+        : m_method(std::move(httpMethod)), m_url(url), m_bodyStream(bodyStream),
           m_retryModeEnabled(false)
     {
     }
 
     // Typically used for GET with no request body.
-    Request(HttpMethod httpMethod, std::string const& url) : Request(httpMethod, url, nullptr) {}
+    Request(HttpMethod httpMethod, std::string const& url)
+        : Request(httpMethod, url, *std::make_unique<NullBodyStream>())
```

Comment:
I don't think this will work, the `std::unique_ptr` will be destructed very soon, then you're referencing a deleted object.

Reply:
This leads to a crash issue.

Reply:
fixed.